### PR TITLE
meson: don't assume libdecaf is present when declaring dependency

### PR DESCRIPTION
**Pull request by omoerbeek**
### Short description

Fixes the build in case libdecaf is not available at all for me (tested on macOS and OpenBSD)

### Checklist

I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [X] compiled this code
- [X] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)
- [ ]  checked that this code was merged to master

### CODE REVIEW

**Comment by zeha** [comment]
unrelated to this fix, but on which platform is '/usr' a valid libdir?

**Comment by fredmorcos** [comment]
> unrelated to this fix, but on which platform is '/usr' a valid libdir?

https://github.com/PowerDNS/pdns/pull/14031#discussion_r1551556290

tldr: I can sleep better at night.

**Review by fredmorcos — APPROVED** [review]
It's good. This is the correct fix.
